src/packet/header/fragment.rs

```diff
@@ -0,0 +1,151 @@
+use super::{HeaderParser, HeaderReader};
+use super::PacketHeader;
+use error::{Result, NetworkError};
+use net::constants::{FRAGMENT_HEADER_SIZE};
+
+use std::io::{self, Cursor, Error, ErrorKind, Write};
+use byteorder::{BigEndian, ReadBytesExt, WriteBytesExt};
+
+#[derive(Copy, Clone, Debug)]
+/// This header represents an fragmented packet header.
+pub struct FragmentHeader {
+    sequence: u16,
+    id: u8,
+    num_fragments: u8,
+    packet_header: Option<PacketHeader>,
+}
+
+impl FragmentHeader {
+    /// Create new fragment with the given packet header
+    pub fn new(id: u8, num_fragments: u8, packet_header: PacketHeader) -> Self {
+        FragmentHeader { id, num_fragments, packet_header: Some(packet_header), sequence: packet_header.seq }
+    }
+
+    /// Get the size of this header.
+    pub fn size(&self) -> u8
```

Comment:
These methods should probably go into a trait.

Reply:
Good point going to do that.